### ISSUE TITLE
SD shortfin CI fails when apt is locked

### ISSUE THREAD

**Issue by renxida** [nod-ai/SHARK-Platform#407]
Should probably first check if cmake & ninja are installed & of the proper version before attempting to install. I don't think checking requires acquiring the apt lock.

![image](https://github.com/user-attachments/assets/1ac25f65-ea73-4cd8-b963-db1957966b87)

**Comment by ScottTodd** [comment]
Workflow history: https://github.com/nod-ai/SHARK-Platform/actions/workflows/ci-sdxl.yaml
Sample logs: https://github.com/nod-ai/SHARK-Platform/actions/runs/11621536945/job/32366319347#step:2:13

Text:
```
Run sudo apt update -y

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
E: Could not get lock /var/lib/apt/lists/lock. It is held by process 2271375 (apt)
E: Unable to lock directory /var/lib/apt/lists/
```

(screenshots aren't searchable of copy/paste-able ;P)

**Comment by renxida** [comment]
good point ty scott!

**Comment by monorimet** [comment]
Added a fix to https://github.com/nod-ai/SHARK-Platform/pull/411, but this still will throw an error if we don't have those installed and cannot acquire the lock. May not be an issue for a while.

output logs:
https://github.com/nod-ai/SHARK-Platform/actions/runs/11637357380/job/32410398564?pr=411#step:2:1

**Comment by ScottTodd** [comment]
> Added a fix to #411

Please keep changes small and send them as individual PRs. We shouldn't need to wait for a larger patch to be reviewed and landed to fix a broken CI build.

Something is suspicious here about the runner environment. Basic dependency setup / installation shouldn't be a sticky source of failures. Might make sense to run these jobs in containers or limit what other code (if any) is running on these machines.

**Comment by monorimet** [comment]
> > Added a fix to #411
> 
> Please keep changes small and send them as individual PRs. We shouldn't need to wait for a larger patch to be reviewed and landed to fix a broken CI build.
> 
> Something is suspicious here about the runner environment. Basic dependency setup / installation shouldn't be a sticky source of failures. Might make sense to run these jobs in containers or limit what other code (if any) is running on these machines.

Agree, will send up separate PR.

Containers are probably a bit heavy for this, but as long as we have env/machine setup flexibility validated within reason, it might be the most robust solution.

**Comment by ScottTodd** [comment]
I'm specifically wondering if `apt update` within a container is free from the hosting system's `/var/lib/apt/lists/lock`. We wouldn't _need_ to use a container for packaging deps (we could)... just for isolation from other users on the runner machine.
